src/walberla_bridge/src/utils/types_conversion.hpp

```diff
@@ -68,6 +68,10 @@ inline Utils::VectorXd<9> to_vector9d(Matrix3<float> const &m) {
           double_c(m[3]), double_c(m[4]), double_c(m[5]),
           double_c(m[6]), double_c(m[7]), double_c(m[8])};
 }
+inline Utils::Vector3i to_vector3i(Vector3<float> const &v) {
+  return Utils::Vector3i{
+      {static_cast<int>(v[0]), static_cast<int>(v[1]), static_cast<int>(v[2])}};
```

Comment:
Are these static casts safe? For example, what is the output when the input vector is `v = {1.f, 2.f, 6.99999f}`? What happens if a vector of doubles is passed? Should we warn the user through assertions if the input floats are not "close enough" to round numbers?

Reply:
I added the function for a vector of doubles and assertions if the difference between input values and round numbers is larger than  10^(-5), which is also debatable.

Reply:
I removed the free function `to_vector3i()` and added the member function of LatticeWalberla `LatticeWalberla::get_block_corner(IBlock &block)`. This change has reduced the scope for using integerisation functions compared to before.